File: 06 -Subprogramas/por_valor_referencia.js
function porValor(x) {
    x += 10;
}

function porReferencia(obj) {
    obj.valor += 10;
}

let a = 5;
let b = { valor: 5 };

porValor(a);
porReferencia(b);

console.log("Valor após porValor:", a);       // 5
console.log("Valor após porReferencia:", b.valor);  // 15

File: 06 -Subprogramas/por_valor_referencia.c
#include <stdio.h>

void porValor(int x) {
    x = x + 10;
}

void porReferencia(int *x) {
    *x = *x + 10;
}

int main() {
    int a = 5, b = 5;
    porValor(a);
    porReferencia(&b);

    printf("Valor após porValor: %d\n", a);       // 5
    printf("Valor após porReferencia: %d\n", b);  // 15
    return 0;
}
